sample/rules.go

```diff
@@ -233,11 +272,20 @@ func extractValueFromSpan(trace *types.Trace, span *types.Span, condition *confi
 	var value any
 	var exists bool
 	for _, field := range condition.Fields {
+		// check if rule uses root span context
+		if strings.HasPrefix(field, RootPrefix) {
+			field = field[len(RootPrefix):]
+			span = trace.RootSpan
+			// if this condition has a root prefix and the root span has already been checked, set flag to true
+			checkedRootPrefix = true
```

Comment:
I don't think this is safe in a multitasking environment. Instead of using a global here, why not return this value from this function as an extra return parameter?

Reply:
Hey kent. Part of the reason that I tried to avoid returning this value from the `extractValueFromSpan` function is because the `trace` and `span` context have to handle these checks differently. I see your point about the global variable not being safe. Do you think that using a variable local to the `ruleMatches...` function would work here? I'll see if I can try pushing something to that effect soon-ish. a

Reply:
I just wanted to elaborate here that my goal with this optimization is to call `extractValueFromSpan` as few times as necessary. Returning this value from that function would necessitate more function calls from the `trace` scope.